File: jgit/com/git/Author.java
package com.git;

import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.time.OffsetDateTime;

class Author {
    private String name;
    private String email;
    private LocalDateTime time;

    Author(String name, String email, LocalDateTime time) {
        this.name = name;
        this.email = email;
        this.time = time;
    }

    public String toString() {
        ZoneOffset offset = OffsetDateTime.now().getOffset();
        Long ltime = time.toEpochSecond(offset);
        return String.format("%s <%s> %d %s", name, email, ltime,
                             offset.toString().replace(":", ""));
    }
}

File: gogit/git/author.go
package git

import "fmt"
import "time"

type Author struct {
	name   string
	email  string
	tstamp time.Time
}

func (author Author) String() string {
	return fmt.Sprintf("%s <%s> %s", author.name, author.email,
		author.tstamp.Format("1257894000 +0300"))
}
